sys/net/application_layer/unicoap/drivers/rfc7252/udp/transport.c

```diff
@@ -0,0 +1,192 @@
+ * Copyright (C) 2024-2025 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @file
+ * @ingroup net_unicoap_drivers_udp
+ * @brief   Transport implementation of CoAP over UDP driver
+ * @author  Carl Seifert <carl.seifert@tu-dresden.de>
+ */
+
+#include <stdint.h>
+#include <errno.h>
+#include "architecture.h"
+#include "net/unicoap/transport.h"
+#include "net/sock.h"
+#include "net/sock/async/types.h"
+#include "net/sock/async/event.h"
+
+#define ENABLE_DEBUG CONFIG_UNICOAP_DEBUG_LOGGING
+#include "debug.h"
+#include "private.h"
+
+#define UDP_DEBUG(...) _UNICOAP_PREFIX_DEBUG(".transport.udp", __VA_ARGS__)
+
+UNICOAP_DECL_RECEIVER_STORAGE_EXTERN;
+
+static sock_udp_t _udp_socket;
+
+extern int unicoap_messaging_process_rfc7252(const uint8_t* pdu, size_t size, bool truncated,
+                                             unicoap_packet_t* packet);
+
+static void _udp_on_event(sock_udp_t* sock, sock_async_flags_t type, void* arg)
+{
+    (void)arg;
+    unicoap_endpoint_t remote = {
+        .proto = UNICOAP_PROTO_UDP,
+    };
+
+    if (type & SOCK_ASYNC_MSG_RECV) {
+        void* stackbuf = NULL;
+        uint8_t* pdu;
+        void* buffer_ctx = NULL;
+        bool truncated = false;
+        ssize_t received = 0;
+
+        sock_udp_aux_rx_t aux_rx = {
+            .flags = IS_ACTIVE(CONFIG_UNICOAP_GET_LOCAL_ENDPOINTS) ? SOCK_AUX_GET_LOCAL : 0,
+        };
+
+        if (IS_ACTIVE(CONFIG_UNICOAP_SOCK_ZERO_COPY_GUARANTEES)) {
+            received = sock_udp_recv_buf_aux(sock, &stackbuf, &buffer_ctx, 0,
+                                             unicoap_endpoint_get_udp(&remote), &aux_rx);
+
+            if (received < 0) {
+                UDP_DEBUG("recv failure: %" PRIdSIZE "\n", received);
+                return;
+            }
+            if (received == 0) {
+                return;
+            }
+            assert(stackbuf);
+            pdu = stackbuf;
+        }
+        else {
+            pdu = unicoap_receiver_buffer;
+            while (true) {
+                ssize_t chunk_size = sock_udp_recv_buf_aux(
+                    sock, &stackbuf, &buffer_ctx, 0, unicoap_endpoint_get_udp(&remote), &aux_rx);
+                if (chunk_size < 0) {
+                    UDP_DEBUG("recv failure: %" PRIdSIZE "\n", chunk_size);
+                    return;
+                }
+                if (chunk_size == 0) {
+                    break;
+                }
+
+                truncated = unicoap_transport_truncate_received((size_t*)&chunk_size, received);
+                memcpy(&pdu[received], stackbuf, chunk_size);
+                received += chunk_size;
+            }
+        }
+
+        if (received == 0) {
+            return;
+        }
+
+        unicoap_packet_t packet = { .remote = &remote };
+
+#     if IS_ACTIVE(CONFIG_UNICOAP_GET_LOCAL_ENDPOINTS)
```

Comment:
```suggestion
#if IS_ACTIVE(CONFIG_UNICOAP_GET_LOCAL_ENDPOINTS)
```

Same with the following `#if` and `#endif`s.